### PR TITLE
Improve unit test TestZeroRequest

### LINKED ISSUE #66468: Improve unit tests of generic_scheduler
**Is this a BUG REPORT or FEATURE REQUEST?**: clean-up

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind feature

Unit tests of generic_scheduler.go includes [a hack](https://github.com/kubernetes/kubernetes/blob/b1e9e747fb777a49ecd204c5e97a5d00d9fcfea1/pkg/scheduler/core/generic_scheduler_test.go#L675) that should be removed. We should add a parameter to the test cases that indicates the expected score and then check if the returned value of the functions matches this expected value.

Below is the hack that we should clean up.

```go
if test.test == "test priority of larger pod with machine with zero-request pod" {
	if hp.Score == expectedPriority {
		t.Errorf("%s: expected non-%d for all priorities, got list %#v", test.test, expectedPriority, list)
	}
}
```

/sig scheduling

### PR DESCRIPTION
**Pull request by islinwb**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66468

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/core/generic_scheduler_test.go [hunk 1]
```diff
@@ -684,12 +687,12 @@ func TestZeroRequest(t *testing.T) {
 			}
 			for _, hp := range list {
 				if test.name == "test priority of larger pod with machine with zero-request pod" {
```
Comment (bsalamat) [inline pkg/scheduler/core/generic_scheduler_test.go]: We actually want to get rid of this `if` statement here. We should not rely on the name of the test. Please remove this if statement and provide the expected score in the corresponding test case.

Comment (islinwb) [reply pkg/scheduler/core/generic_scheduler_test.go]: @bsalamat Sorry, I thougt the score of the third case varies.

**Review by bsalamat — APPROVED** [review]
/lgtm

**Pull request merged**